### PR TITLE
Support jinja templates for sql files in clickhouse-test

### PR DESCRIPTION
**Pull request by vdimir**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

`clickhouse-test` supports SQL tests with [Jinja2](https://jinja.palletsprojects.com/en/3.0.x/templates/#synopsis) templates

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]
Ok.

Let's confirm: I can use `.reference` or `.reference.j2`, whatever I prefer?

File: tests/clickhouse-test [hunk 1]
```diff
@@ -760,6 +771,97 @@ def do_run_tests(jobs, suite, suite_dir, suite_tmp_dir, all_tests, parallel_test
...
+
+def get_reference_file(suite_dir, name):
+    """
+    Returns reference file name for specified test
+    """
+
+    name = removesuffix(name, ".gen")
+    for ext in ['.reference', '.gen.reference']:
+        reference_file = os.path.join(suite_dir, name) + ext
+        if os.path.isfile(reference_file):
```
Comment (alexey-milovidov) [inline tests/clickhouse-test:860]: Ok.

File: tests/clickhouse-test [hunk 2]
```diff
@@ -760,6 +771,97 @@ def do_run_tests(jobs, suite, suite_dir, suite_tmp_dir, all_tests, parallel_test
...
+    # as a query editor in the test, and must be marked as executable.
+    return os.path.isfile(case_file) and any(case_file.endswith(suppotred_ext) for suppotred_ext in TEST_FILE_EXTENSIONS)
+
+
+def removesuffix(text, *suffixes):
+    """
+    Added in python 3.9
+    https://www.python.org/dev/peps/pep-0616/
+
+    This version can work with severtal possible suffixes
```
Comment (nikitamikhaylov) [inline tests/clickhouse-test:787]:
```suggestion
    This version can work with several possible suffixes
```

**Review by nikitamikhaylov — APPROVED** [review]

File: tests/clickhouse-test (hunk 2 above)
Comment (alexey-milovidov) [reply tests/clickhouse-test:787]: Oops.

Let's commit this change to master.

**Comment by vdimir** [comment]
> I can use `.reference` or `.reference.j2`, whatever I prefer?

Yes, even if sql test file is template (`.sql.j2`) corresponding reference can be either template (`.reference.j2`) or not (`.reference`)

**Pull request merged**